listenbrainz_spark/recommendations/create_dataframes.py

```diff
@@ -160,30 +158,36 @@ def get_listens_for_training_model_window(to_date, from_date, metadata, dest_pat
     return partial_listens_df
 
 
-def get_mapped_artist_and_recording_mbids(partial_listens_df, recording_artist_mapping_df):
+def get_mapped_artist_and_recording_mbids(partial_listens_df, msid_mbid_mapping_df):
     """ Map recording msid->mbid and artist msid->mbids so that every listen has an mbid.
 
         Args:
             partial_listens_df (dataframe): listens without artist mbid and recording mbid.
-            recording_artist_mapping_df (dataframe): msid->mbid mapping. For columns refer to
-                                                     msid_mbid_mapping_schema in listenbrainz_spark/schema.py
+            msid_mbid_mapping_df (dataframe): msid->mbid mapping. For columns refer to
+                                              msid_mbid_mapping_schema in listenbrainz_spark/schema.py
 
         Returns:
-            mapped_listens (dataframe): listens mapped with msid_mbid_mapping.
+            mapped_listens_df (dataframe): listens mapped with msid_mbid_mapping.
     """
-    df = partial_listens_df.join(recording_artist_mapping_df,
-            (partial_listens_df.recording_msid == recording_artist_mapping_df.msb_recording_msid) &
-            (partial_listens_df.artist_msid == recording_artist_mapping_df.msb_artist_msid),
-        'inner'
-    )
+    condition = [
+        partial_listens_df.recording_msid == msid_mbid_mapping_df.msb_recording_msid,
+        partial_listens_df.artist_msid == msid_mbid_mapping_df.msb_artist_msid
+    ]
+
+    df = partial_listens_df.join(msid_mbid_mapping_df, condition, 'inner')
     # msb_release_name_matchable is skipped till the bug in mapping is resolved.
```

Comment:
This bug should now be fixed. Can you please confirm?

Reply:
I don't have the latest mapping on my local machine (and I cannot download for the size) :( 

I will ask iliekcomputers to run a test query for this. But if you have removed the bug I can remove the comment.

Reply:
You could download the mapping into your account on newleader and check there....

Reply:
Oh like check manually? Cool, will do.

Reply:
should I keep the comment and merge this PR since we found a few bugs yesterday?

Reply:
I pushed a new mapping. Have you verified that the new mapping contains further bugs?